doc/lua_api.txt

```diff
@@ -3702,8 +3702,8 @@ Call these functions only at load time!
     * Called when an item is eaten, by `minetest.item_eat`
     * Return `true` or `itemstack` to cancel the default item eat response
       (i.e.: hp increase).
-* `minetest.register_on_item_pickup(function(luaentity, picker, time_from_last_punch, tool_capabilities, dir))`
-    * Called when item is picked up by `picker`.
+* `minetest.register_on_item_pickup(function(luaentity, inventory, picker, ...))`
+    * Called when an item is picked up, by `minetest.item_pickup`.
```

Comment:
Use this instead:
```lua
default: `minetest.item_pickup`
```

See [here](https://github.com/minetest/minetest/blob/a5197eaebc61ac6b555a640f36c0b427faef381d/doc/lua_api.txt#L5793) and [here](https://github.com/minetest/minetest/blob/a5197eaebc61ac6b555a640f36c0b427faef381d/doc/lua_api.txt#L5803) for how other callbacks document the default functions.

Reply:
But it is not the default.
`minetest.item_pickup` calls the callbacks of `minetest.register_on_item_pickup` and item def.
It's about the same as above (item_eat).

Reply:
Ah, I see now. Would it make sense to **make** `minetest.item_pickup` the default, which would be omitted when `on_pickup` has been defined for the item? To add some custom behaviour _alongside_ the actual picking up behaviour, the item can define a custom `on_pickup` callback, that calls `minetest.item_pickup` within. This implementation would be consistent with how the other itemdef callbacks work.

Reply:
`minetest.item_pickup` is for mods that want to pick up an item (eg. vacuuming nodes).
I don't think, it would make sense. By replacing the default you would probably omit the `registered_on_item_pickup` callbacks. Instead of calling a function in the callback, you can simply return `true`.

Reply:
The `registered_on_pickup` callbacks can be directly run from `on_punch` of the item entity. This is currently how other callbacks are handled. e.g. digging a node will run all `registered_on_dignode` callbacks, but will run `minetest.dig_node` only if the `on_dig` callback hasn't been defined.

Reply:
Many callbacks use two functions, eg. eat (`item_eat` returns a function that uses `do_item_eat`), dig (`dig_node` calls `node_dig`, `node_dig` is like `item_pickup` and removes the node directly[()](https://github.com/minetest/minetest/blob/5560ec8178053b27adac6f277f539975d46267b8/builtin/game/item.lua#L543)).

I think, there are already many inconsistencies.

[Digging a node](https://github.com/minetest/minetest/blob/3d66622772e66154b7624957a27f9be54c4c7c28/src/network/serverpackethandler.cpp#L1285) [always calls the `on_dig` callback](https://github.com/minetest/minetest/blob/eef62c82a2e58700fc1216b0b8c03e421bc77995/src/script/cpp_api/s_node.cpp#L129) [which by default (`node_dig`) does everything (calling callbacks (including `can_dig` which can be compared to `on_pickup`), removing the node, …).](https://github.com/minetest/minetest/blob/5560ec8178053b27adac6f277f539975d46267b8/builtin/game/item.lua#L543) It has to be done like this because digging nodes comes from c++. [`dig_node` is only used by lua mods.](https://github.com/minetest/minetest/blob/5f1cd555cd9d1c64426e173b30b5b792d211c835/src/script/lua_api/l_env.cpp#L456) [The `registered_on_dignodes` callbacks are not always called since they are called by `node_dig`.](https://github.com/minetest/minetest/blob/5560ec8178053b27adac6f277f539975d46267b8/builtin/game/item.lua#L621-L637)

Btw. should I set the mod origin?